app/components/Form/AmountInput.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import CurrencyIcon from 'components/CurrencyIcon'
+import styles from './RequestForm.scss'
+
+class AmountInput extends React.Component {
+
+  constructor(props) {
+    super(props)
+
+    this.handleChange = this.handleChange.bind(this)
+    this.handleKeyDown = this.handleKeyDown.bind(this)
+  }
+
+  parseNumber(_value) {
+    let value = _value.replace(/[^0-9,.]/g, '')
+    let integer = null
+    let fractional = null
+
+    if (value == '') { value = '0' }
+    if (value*1.0 < 0) { value = '0.0' }
+
+    // simplify this, make flat if possible
+    // extract the parsing bit?
+    if (value.match(/^[0-9]*[.,][0-9]*$/)) {
+      [integer, fractional] = value.toString().split(/[.,]/)
+      if (!fractional || fractional == '') {
+        fractional = ''
+      }
+    } else {
+      integer = value
+    }
+
+    // limit fractional precision to 8 places
+    if (fractional && fractional.length > 8) {
+      fractional = fractional.substring(0, 8)
+    }
+
+    return [integer, fractional]
+  }
+
+  formatValue(integer, fractional) {
+    let value;
+    if (fractional && fractional.length > 0) {
+      value = `${integer}.${fractional}`
+    } else { 
+      if (fractional === '') {
```

Comment:
this conditional sets value = `${integer}.` no matter what. doesn't look like it's needed

Reply:
When integer is null, it runs the else part. Thats how it knows its just `X.` vs just `X`